File: GUI/seminar/src/app/models/Catalog.ts
import { CatalogItem } from "./CatalogItem";
import { Company } from "./Company";

export class Catalog {
    id: number;
    name: string;
    description: string = "";
    companyId: number;
    company!: Company;
    companyName: string = "";
    catalogItems!: CatalogItem;

    catalogTypeId: number = 0;
    catalogTypeName: string = "";
    isEnabled: boolean;
    
    created!: Date;
    createdUser: number  = 0;
    modified!: Date;
    modifiedUser: number = 0;


    constructor() {
        this.id = 0;
        this.name = '';
        this.companyId = 0;
        this.companyName = "";
        this.company = new Company();
        this.isEnabled = false;
        this.description = '';
    }
}

File: API/domain/Entities/realestate/Catalog.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using domain.Entities.common;
using domain.Entities.crm;

namespace domain.Entities.releastate
{
    public class Catalog : BaseEntity
    {
        public Catalog()
        {
            CatalogItems = new HashSet<CatalogItem>();
            CatalogPeriods = new HashSet<CatalogPeriod>();
        }
        //Database Atributes
        public int CompanyId { get; set; }
        public string? Name { get; set; }
        public string? Quote { get; set; }
        public string? Description { get; set; }
        public int? CatalogTypeId {get;set;}
        public bool? IsEnabled { get; set; }


        //Referent tables
        public virtual Company Company { get; set; }
        public virtual CatalogType CatalogType { get; set; }

        //Table Sets
        public virtual ICollection<CatalogItem> CatalogItems { get; set; }
        public virtual ICollection<CatalogPeriod> CatalogPeriods { get; set; }
    }
}
